ccl/utilccl: TestRefreshLicenseEnforcerOnLicenseChange failed

ccl/utilccl.TestRefreshLicenseEnforcerOnLicenseChange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/17506215?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/17506215?buildTab=artifacts#/) on release-23.1 @ [2aee3018178db60f0e909ced0a2d1a9148b869e5](https://github.com/cockroachdb/cockroach/commits/2aee3018178db60f0e909ced0a2d1a9148b869e5):


```
I241030 09:53:17.703970 27060 upgrade/upgrademanager/manager.go:278  [T1,n1] 166  running permanent upgrade for version 22.2-102
I241030 09:53:17.789260 27060 upgrade/upgrademanager/manager.go:737  [T1,n1] 167  running Upgrade to 22.2-102: "create sql activity updater job"
I241030 09:53:17.789709 384445 jobs/adopt.go:256  [T1,n1] 168  job 1016501363356598273: resuming execution
I241030 09:53:17.858315 386728 jobs/registry.go:1623  [T1,n1] 169  MIGRATION job 1016501363356598273: stepping through state running
I241030 09:53:17.979858 386728 jobs/registry.go:1623  [T1,n1] 170  MIGRATION job 1016501363356598273: stepping through state succeeded
I241030 09:53:18.077014 27060 jobs/wait.go:140  [T1,n1] 171  waited for 1 [1016501363356598273] queued jobs to complete 287.56804ms
I241030 09:53:18.077126 27060 server/server_sql.go:1633  [T1,n1] 172  done ensuring all necessary startup migrations have run
I241030 09:53:18.080529 27060 server/license/enforcer.go:233  [T1,n1] 173  trial license expiry initialized to 1970-01-01 00:00:00 +0000 UTC
I241030 09:53:18.081260 403460 jobs/job_scheduler.go:402  [T1,n1] 174  waiting 4m0s before scheduled jobs daemon start
I241030 09:53:18.083846 37804 kv/kvserver/replica_rangefeed.go:755  [T1,n1,s1,r22/1:/Table/2{0-1}] 175  RangeFeed closed timestamp is empty
I241030 09:53:18.099796 27060 server/license/enforcer.go:630  [T1,n1] 176  cluster init is not within the bounds of the enforcer start time: 2024-10-30 09:53:13.67719 +0000 UTC
I241030 09:53:18.101306 27060 server/license/enforcer.go:264  [T1,n1] 177  generated new cluster init grace period end time: 2024-09-21 12:28:36 +0000 UTC
I241030 09:53:18.117099 27060 server/license/enforcer.go:447  [T1,n1] 178  enforcer license updated: type is none, grace period ends at "2024-09-21 12:28:36 +0000 UTC", telemetry required: false
I241030 09:53:18.117230 27060 server/license/enforcer.go:447  [T1,n1] 179  enforcer license updated: type is none, grace period ends at "2024-09-21 12:28:36 +0000 UTC", telemetry required: false
I241030 09:53:18.122168 408942 kv/kvclient/rangefeed/rangefeedcache/watcher.go:330  [T1,n1] 180  tenant-settings-watcher: established range feed cache
I241030 09:53:18.133751 403470 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 181  warmed privileges for virtual tables in 52.062808ms
I241030 09:53:18.142380 410946 kv/kvclient/rangefeed/rangefeedcache/watcher.go:330  [T1,n1] 182  tenant-capability-watcher: established range feed cache
I241030 09:53:18.149515 408634 server/auto_upgrade.go:73  [T1,n1] 183  no need to upgrade, cluster already at the newest version
I241030 09:53:18.150780 410946 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:144  [T1,n1] 184  received results of a full table scan for tenant capabilities
I241030 09:53:18.201344 37278 3@pebble/event.go:717  [n1,s1,pebble] 185  [JOB 7] WAL created 000006
I241030 09:53:18.259545 417252 1@util/log/event_log.go:27  [intExec=set-zone] 186 ={"Timestamp":1730281998246826635,"EventType":"set_zone_config","Statement":"ALTER RANGE default CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","ApplicationName":"$ internal-set-zone","Target":"RANGE default","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:14400 > num_replicas:1 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.295127 416465 1@util/log/event_log.go:27  [intExec=set-zone] 187 ={"Timestamp":1730281998270740390,"EventType":"set_zone_config","Statement":"ALTER DATABASE system CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":1,"ApplicationName":"$ internal-set-zone","Target":"DATABASE system","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:14400 > num_replicas:5 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.322492 421237 1@util/log/event_log.go:27  [intExec=set-zone] 188 ={"Timestamp":1730281998297031974,"EventType":"set_zone_config","Statement":"ALTER RANGE meta CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":16,"ApplicationName":"$ internal-set-zone","Target":"RANGE meta","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:3600 > num_replicas:5 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.359059 425356 1@util/log/event_log.go:27  [intExec=set-zone] 189 ={"Timestamp":1730281998343719094,"EventType":"set_zone_config","Statement":"ALTER RANGE system CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":17,"ApplicationName":"$ internal-set-zone","Target":"RANGE system","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:14400 > num_replicas:5 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.396562 426118 1@util/log/event_log.go:27  [intExec=set-zone] 190 ={"Timestamp":1730281998362033492,"EventType":"set_zone_config","Statement":"ALTER RANGE liveness CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":22,"ApplicationName":"$ internal-set-zone","Target":"RANGE liveness","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:600 > num_replicas:5 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.432612 429643 1@util/log/event_log.go:27  [intExec=set-zone] 191 ={"Timestamp":1730281998400865791,"EventType":"set_zone_config","Statement":"ALTER TABLE system.public.replication_constraint_stats CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":25,"ApplicationName":"$ internal-set-zone","Target":"TABLE system.public.replication_constraint_stats","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:600 > num_replicas:1 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.451967 432358 1@util/log/event_log.go:27  [intExec=set-zone] 192 ={"Timestamp":1730281998435885978,"EventType":"set_zone_config","Statement":"ALTER TABLE system.public.replication_stats CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":27,"ApplicationName":"$ internal-set-zone","Target":"TABLE system.public.replication_stats","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:600 > num_replicas:1 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.589571 441422 1@util/log/event_log.go:27  [intExec=set-zone] 193 ={"Timestamp":1730281998558290318,"EventType":"set_zone_config","Statement":"ALTER TABLE system.public.statement_statistics CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-zone","Target":"TABLE system.public.statement_statistics","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:3600 > num_replicas:1 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.614417 442745 1@util/log/event_log.go:27  [intExec=set-zone] 194 ={"Timestamp":1730281998592353583,"EventType":"set_zone_config","Statement":"ALTER TABLE system.public.transaction_statistics CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-zone","Target":"TABLE system.public.transaction_statistics","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:3600 > num_replicas:1 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.636530 445445 1@util/log/event_log.go:27  [intExec=set-zone] 195 ={"Timestamp":1730281998619105139,"EventType":"set_zone_config","Statement":"ALTER TABLE system.public.tenant_usage CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":45,"ApplicationName":"$ internal-set-zone","Target":"TABLE system.public.tenant_usage","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:7200 > num_replicas:1 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.674431 447652 1@util/log/event_log.go:27  [intExec=set-zone] 196 ={"Timestamp":1730281998639333373,"EventType":"set_zone_config","Statement":"ALTER TABLE system.public.span_stats_tenant_boundaries CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":57,"ApplicationName":"$ internal-set-zone","Target":"TABLE system.public.span_stats_tenant_boundaries","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:3600 > num_replicas:1 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.696994 449262 1@util/log/event_log.go:27  [intExec=set-zone] 197 ={"Timestamp":1730281998676733896,"EventType":"set_zone_config","Statement":"ALTER TABLE system.public.statement_activity CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-zone","Target":"TABLE system.public.statement_activity","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:3600 > num_replicas:1 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.721706 453623 1@util/log/event_log.go:27  [intExec=set-zone] 198 ={"Timestamp":1730281998700817221,"EventType":"set_zone_config","Statement":"ALTER TABLE system.public.transaction_activity CONFIGURE ZONE USING num_replicas = 1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-zone","Target":"TABLE system.public.transaction_activity","Options":["num_replicas = 1"],"ResolvedOldConfig":"range_min_bytes:134217728 range_max_bytes:536870912 gc:<ttl_seconds:3600 > num_replicas:1 inherited_constraints:false null_voter_constraints_is_empty:true inherited_lease_preferences:false "}
I241030 09:53:18.728927 27060 1@server/initial_sql.go:42  [-] 199  Replication was disabled for this cluster.
I241030 09:53:18.728927 27060 1@server/initial_sql.go:42  [-] 199 +When/if adding nodes in the future, update zone configurations to increase the replication factor.
I241030 09:53:18.729027 27060 1@server/server_sql.go:1781  [T1,n1] 200  serving sql connections
I241030 09:53:18.735046 457921 sql/pgwire/pre_serve.go:515  [T1,n1,client=127.0.0.1:57396] 201  sending sslSupported message to client
I241030 09:53:18.746599 459094 4@util/log/event_log.go:27  [T1,n1,client=127.0.0.1:57396,hostssl,user=root] 202 ={"Timestamp":1730281998746595480,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:57396","SessionID":"1803324f7883caed0000000000000001","Transport":"hostssl","User":"root","SystemIdentity":"root","Method":"cert-password"}
I241030 09:53:18.762747 459094 util/log/event_log.go:27  [T1,n1,client=127.0.0.1:57396,hostssl,user=root] 203 ={"Timestamp":1730281998746980270,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.organization\" = 'CRDB Unit Test'","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"cluster.organization","Value":"'CRDB Unit Test'"}
--- FAIL: TestRefreshLicenseEnforcerOnLicenseChange (6.40s)
=== RUN   TestRefreshLicenseEnforcerOnLicenseChange/test_2
I241030 09:53:18.820201 51824 server/license/enforcer.go:485  [T1,n1,rangefeed=settings-watcher] 210  trial license expiry timestamp is 2000-01-01 12:00:00 +0000 UTC
    license_check_test.go:274: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/utilccl/license_check_test.go:274
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 946728000
        	Test:       	TestRefreshLicenseEnforcerOnLicenseChange/test_2
I241030 09:53:18.833177 466602 server/license/enforcer.go:501  [-] 211  trial license expiry was reset
    --- FAIL: TestRefreshLicenseEnforcerOnLicenseChange/test_2 (0.01s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefreshLicenseEnforcerOnLicenseChange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-43773

--- a/pkg/ccl/utilccl/license_check_test.go
+++ b/pkg/ccl/utilccl/license_check_test.go
@@ -292,7 +292,6 @@ func TestRefreshLicenseEnforcerOnLicenseChange(t *testing.T) {
 				require.NoError(t, err)
 				trialLicenseExpiryTimestamp.Store(0)
 			}()
-			require.Equal(t, int64(0), trialLicenseExpiryTimestamp.Load())
 
 			tdb := sqlutils.MakeSQLRunner(sqlDB)
 

